types/time.go

```diff
@@ -562,165 +564,444 @@ func ParseDateFormat(format string) []string {
 	format = strings.TrimSpace(format)
 
 	start := 0
-	var seps = make([]string, 0)
-	for i := 0; i < len(format); i++ {
-		// Date format must start and end with number.
-		if i == 0 || i == len(format)-1 {
-			if !unicode.IsNumber(rune(format[i])) {
-				return nil
-			}
-
-			continue
+	var seps = make([]string, 0, maxSQLTimeParts)
+	if len(format) == 0 {
+		return nil
+	}
+	// Date format must start with number.
+	if !unicode.IsNumber(rune(format[0])) {
```

Comment:
Should end with number too.

Reply:
But I find that it works well although not end with number.
`
mysql> Select cast('1701020302.11BB' as datetime); 
+-------------------------------------+
| cast('1701020302.11BB' as datetime) |
+-------------------------------------+
| 2017-01-02 03:02:11                 |
+-------------------------------------+
1 row in set, 1 warning (0.00 sec)
`

Reply:
Ummm, that's true.